test/distributed/_composable/test_replicate_with_compiler.py

```diff
@@ -318,7 +318,11 @@ def bwd(loss):
     # todo: This pass mucks things up since Inductor thinks its inference
     # and can apply this. Should turn off these passes in compiled autograd
     @torch._inductor.config.patch(
-        reorder_for_locality=False, reorder_for_peak_memory=False
+        reorder_for_locality=False,
+        reorder_for_peak_memory=False,
+        # The correctness of this test relies on the pointless permute ops
+        # in the joint graph does not get eliminated..
+        pattern_matcher=False,
```

Comment:
cc @fegin

Reply:
You probably can change what it is checking for, to match newly generated code.

Reply:
Changing L332 to loop only twice can make the test pass. But I also see the code checks that 'ddp_buckets' equals to 3, so I'm not sure if I'll make the test less helpful by doing that.